rmw_connextdds_common/src/common/rmw_publication.cpp

```diff
@@ -141,6 +142,10 @@ rmw_api_connextdds_create_publisher(
   RMW_CHECK_ARGUMENT_FOR_NULL(qos_policies, nullptr);
 #if RMW_CONNEXT_HAVE_OPTIONS_PUBSUB
   RMW_CHECK_ARGUMENT_FOR_NULL(publisher_options, nullptr);
+  if (publisher_options->require_unique_network_flow_endpoints == RMW_UNIQUE_NETWORK_FLOW_ENDPOINTS_STRICTLY_REQUIRED) {
+    RMW_CONNEXT_LOG_ERROR_SET("Strict requirement on unique network flow endpoints for publishers not supported");
+    return nullptr;
+  }
```

Comment:
As you probably saw, this implementation defines a series of "feature flags" in `static_config.hpp` to be able to support multiple versions of ROS2 from the same branch. This is to avoid having to create branches for different versions, which we'll probably end up doing starting with Galactic, but would have been a pain to maintain up until now, while the RMW was in very active development.

For now, I'd like to continue to adhere with this pattern so that the `master` branch remains buildable with Foxy, so I suggest introducing a new flag in `static_config.hpp` to make sure field `require_unique_network_flow_endpoints` is only accessed in the appropriate ROS2 version. Something like:

```cpp
#ifndef RMW_CONNEXT_HAVE_UNIQUE_NETWORK_FLOWS
#define RMW_CONNEXT_HAVE_UNIQUE_NETWORK_FLOWS \
  (RMW_CONNEXT_RELEASE > RMW_CONNEXT_RELEASE_FOXY)
#endif /* RMW_CONNEXT_HAVE_UNIQUE_NETWORK_FLOWS */
```

Then wrap the new code in an `#if RMW_CONNEXT_HAVE_UNIQUE_NETWORK_FLOWS` (also in `rmw_subscription.cpp`).